x-pack/plugins/infra/public/utils/logs_overview_fetchers.ts

```diff
@@ -4,90 +4,179 @@
  * you may not use this file except in compliance with the Elastic License.
  */
 
-import { InfraClientCoreSetup } from '../types';
-import { LogsFetchDataResponse } from '../../../observability/public';
+import { InfraClientCoreSetup, InfraClientStartDeps } from '../types';
+import {
+  FetchData,
+  LogsFetchDataResponse,
+  HasData,
+  FetchDataParams,
+} from '../../../observability/public';
+import { callFetchLogSourceConfigurationAPI } from '../containers/logs/log_source/api/fetch_log_source_configuration';
+import { callFetchLogSourceStatusAPI } from '../containers/logs/log_source/api/fetch_log_source_status';
 
-export function getLogsHasDataFetcher(getStartServices: InfraClientCoreSetup['getStartServices']) {
-  return async () => {
-    // if you need the data plugin, this is how you get it
-    // const [, startPlugins] = await getStartServices();
-    // const { data } = startPlugins;
+interface StatsAggregation {
+  buckets: Array<{ key: string; doc_count: number }>;
+}
 
-    // if you need a core dep, we need to pass in more than just getStartServices
+interface SeriesAggregation {
+  buckets: Array<{
+    key_as_string: string;
+    key: number;
+    doc_count: number;
+    dataset: StatsAggregation;
+  }>;
+}
+
+interface LogParams {
+  index: string;
+  timestampField: string;
+}
 
-    // perform query
-    return true;
+type StatsAndSeries = Pick<LogsFetchDataResponse, 'stats' | 'series'>;
+
+const SOURCE_ID = 'default';
```

Comment:
Can this be specified in some common constant file and imported? I'm nervous we are going to flub this up somewhere accidentally...

Reply:
I just checked if we happen to have a constant already for this, but we use the hardcoded `'default'` in a lot of places.

<pre><details><summary>
% git grep "'default'" -- x-pack/plugins/infra</summary>
x-pack/plugins/infra/public/alerting/inventory/components/expression.tsx:103:    sourceId: 'default',
x-pack/plugins/infra/public/alerting/inventory/components/expression.tsx:305:      setAlertParams('sourceId', source?.id || 'default');
x-pack/plugins/infra/public/alerting/metric_threshold/components/expression.test.tsx:22:    source: { id: 'default' },
x-pack/plugins/infra/public/alerting/metric_threshold/components/expression.tsx:82:    sourceId: 'default',
x-pack/plugins/infra/public/alerting/metric_threshold/components/expression.tsx:295:      setAlertParams('sourceId', source?.id || 'default');
x-pack/plugins/infra/public/alerting/metric_threshold/components/expression_chart.test.tsx:58:      id: 'default',
x-pack/plugins/infra/public/alerting/metric_threshold/components/expression_chart.test.tsx:61:        name: 'default',
x-pack/plugins/infra/public/alerting/metric_threshold/components/expression_chart.test.tsx:108:      sourceId: 'default',
x-pack/plugins/infra/public/containers/source/use_source_via_http.ts:45:  sourceId = 'default',
x-pack/plugins/infra/public/containers/source_id/source_id.ts:18:    defaultState: 'default',
x-pack/plugins/infra/public/pages/link_to/redirect_to_host_detail_via_ip.tsx:28:  const { source } = useSource({ sourceId: 'default' });
x-pack/plugins/infra/public/pages/link_to/redirect_to_logs.tsx:25:  const sourceId = match.params.sourceId || 'default';
x-pack/plugins/infra/public/pages/link_to/redirect_to_node_logs.tsx:40:    params: { nodeId, nodeType, sourceId = 'default' },
x-pack/plugins/infra/public/pages/metrics/index.tsx:180:      shouldLoadDefault={options.source === 'default'}
x-pack/plugins/infra/public/pages/metrics/inventory_view/hooks/use_waffle_options.ts:42:  source: 'default',
x-pack/plugins/infra/public/pages/metrics/inventory_view/index.tsx:67:              shouldLoadDefault={optionsSource === 'default'}
x-pack/plugins/infra/public/pages/metrics/metrics_explorer/components/chart_title.tsx:23:              <EuiText size="m" color={keys.length - 1 > i ? 'subdued' : 'default'}>
x-pack/plugins/infra/public/pages/metrics/metrics_explorer/hooks/use_metrics_explorer_options.ts:88:  source: 'default',
x-pack/plugins/infra/public/utils/fixtures/metrics_explorer.ts:27:  name: 'default',
x-pack/plugins/infra/public/utils/styles.ts:31:    default: Symbol('default'),
x-pack/plugins/infra/public/utils/use_kibana_space_id.ts:21:      () => 'default',
x-pack/plugins/infra/server/lib/adapters/framework/kibana_framework_adapter.ts:240:      return 'default';
x-pack/plugins/infra/server/lib/alerting/inventory_metric_threshold/inventory_metric_threshold_executor.ts:31:    sourceId || 'default'
x-pack/plugins/infra/server/lib/alerting/log_threshold/log_threshold_executor.ts:43:    const sourceConfiguration = await sources.getSourceConfiguration(savedObjectsClient, 'default');
x-pack/plugins/infra/server/lib/alerting/metric_threshold/metric_threshold_executor.test.ts:28:    const execute = (comparator: Comparator, threshold: number[], sourceId: string = 'default') =>
x-pack/plugins/infra/server/lib/alerting/metric_threshold/metric_threshold_executor.test.ts:413:  return { id: 'default', attributes: { metricAlias: 'metricbeat-*' }, type, references: [] };
x-pack/plugins/infra/server/lib/alerting/metric_threshold/metric_threshold_executor.ts:31:      sourceId || 'default'
x-pack/plugins/infra/server/plugin.ts:161:        const spaceId = plugins.spaces?.spacesService.getSpaceId(request) || 'default';
x-pack/plugins/infra/server/routes/alerting/preview.ts:41:        sourceId || 'default'
</details></pre>

I can create a `infra/common/constants.ts` file in this PR, and later open another one that replaces the relevant hardcoded strings to that new constant.